pynotify/handlers.py

```diff
@@ -23,7 +23,7 @@ def __new__(cls, name, bases, attrs):
                 # delete the abstract attribute so it is not inherited by child classes
                 delattr(new_class.Meta, 'abstract')
             else:
-                if not getattr(new_class, 'Meta', None) or not getattr(new_class.Meta, 'signal', None):
+                if not getattr(new_class, 'Meta', None) or not hasattr(new_class.Meta, 'signal'):
```

Comment:
Newly, the signal might not be set, but that must be explicitly stated using `signal = None`. In that case, the handler won't be registered and can be called using some custom method.